--- /Users/ybao/asterisk-15.1.2-src/res/res_format_attr_h264.c	2018-01-23 10:33:38.000000000 -0500
+++ res_format_attr_h264_patched2.c	2018-02-22 13:37:06.000000000 -0500
@@ -210,9 +210,9 @@
 			attr->REDUNDANT_PIC_CAP = val;
 		} else if (sscanf(attrib, "parameter-add=%30u", &val) == 1) {
 			attr->PARAMETER_ADD = val;
-		} else if (sscanf(attrib, "packetization-mode=%30u", &val) == 1) {
+	/*	} else if (sscanf(attrib, "packetization-mode=%30u", &val) == 1) {
 			attr->PACKETIZATION_MODE = val;
-		} else if (sscanf(attrib, "sprop-interleaving-depth=%30u", &val) == 1) {
+	*/	} else if (sscanf(attrib, "sprop-interleaving-depth=%30u", &val) == 1) {
 			attr->SPROP_INTERLEAVING_DEPTH = val;
 		} else if (sscanf(attrib, "sprop-deint-buf-req=%30u", &val) == 1) {
 			attr->SPROP_DEINT_BUF_REQ = val;
@@ -263,6 +263,21 @@
 		return;
 	}
 
+
+//Start Yin's change - conditional SDP mod only if profile level is high, also don't overwrite rate parameters, use profile 0x42001f per Chrome64
+	if (attr->PROFILE_IDC == 0x64) {
+        //attr->MAX_MBPS = 40500;
+        //attr->MAX_FS = 1344;
+        //attr->MAX_BR = 906;
+        //attr->MAX_SMBPS = 40500;
+        //attr->MAX_FPS = 3000;
+        attr->PROFILE_IDC = 0x42;
+        attr->PROFILE_IOP = 0x00;
+        attr->LEVEL = 0x1f;
+	}
+//End Yin's change
+
+
 	APPEND_IF_NONZERO(attr->MAX_MBPS, str, "max-mbps");
 	APPEND_IF_NONZERO(attr->MAX_FS, str, "max-fs");
 	APPEND_IF_NONZERO(attr->MAX_CPB, str, "max-cpb");
@@ -279,10 +294,22 @@
 
 	APPEND_IF_NOT_H264_UNSET(attr->REDUNDANT_PIC_CAP, str, "redundant-pic-cap");
 	APPEND_IF_NOT_H264_UNSET(attr->PARAMETER_ADD, str, "parameter-add");
-	APPEND_IF_NOT_H264_UNSET(attr->PACKETIZATION_MODE, str, "packetization-mode");
+//	APPEND_IF_NOT_H264_UNSET(attr->PACKETIZATION_MODE, str, "packetization-mode");
 	APPEND_IF_NOT_H264_UNSET(attr->LEVEL_ASYMMETRY_ALLOWED, str, "level-asymmetry-allowed");
 
-	if (attr->PROFILE_IDC && attr->PROFILE_IOP && attr->LEVEL) {
+
+        //Start force packetization-mode=1 for a sorenson test
+        ast_log(LOG_NOTICE, "NOTICE=>Adding packetization str \n");
+        if (added) {
+                ast_str_append(str, 0, ";");
+        } else if (0 < ast_str_append(str, 0, "a=fmtp:%u ", payload)) {
+                added = 1;
+        }
+        ast_str_append(str, 0, "packetization-mode=%s", "1");
+       //End force packetization-mode=1 for a sorenson test
+
+	if (attr->PROFILE_IDC | attr->PROFILE_IOP | attr->LEVEL) { 
+/*	if (attr->PROFILE_IDC && attr->PROFILE_IOP && attr->LEVEL) { */
 		if (added) {
 			ast_str_append(str, 0, ";");
 		} else if (0 < ast_str_append(str, 0, "a=fmtp:%u ", payload)) {
