--- res_musiconhold.c.original	2018-02-22 18:16:01.052745484 +0000
+++ res_musiconhold.c	2018-02-22 18:24:52.644381035 +0000
@@ -190,6 +190,8 @@
 	enum kill_methods kill_method;
 	/*! Source of audio */
 	int srcfd;
+	/*! Flag to indicate if this is a video class */
+	int video;
 	/*! Generic timer */
 	struct ast_timer *timer;
 	/*! Created on the fly, from RT engine */
@@ -366,7 +368,13 @@
 	}
 
 	for (tries = 0; tries < state->class->total_files; ++tries) {
-		if (ast_openstream_full(chan, state->class->filearray[state->pos], ast_channel_language(chan), 1)) {
+
+		int v = state->class->video;
+		
+
+		if(v && ast_openvstream(chan, state->class->filearray[state->pos], ast_channel_language(chan)) && ast_openstream_full(chan, state->class->filearray[state->pos], ast_channel_language(chan), 1)){
+			break;
+		} else if (ast_openstream_full(chan, state->class->filearray[state->pos], ast_channel_language(chan), 1)) {
 			break;
 		}
 
@@ -402,11 +410,20 @@
 
 static struct ast_frame *moh_files_readframe(struct ast_channel *chan)
 {
+	struct moh_files_state *state = ast_channel_music_state(chan);
 	struct ast_frame *f = NULL;
+	int v = state->class->video;
 
-	if (!(ast_channel_stream(chan) && (f = ast_readframe(ast_channel_stream(chan))))) {
-		if (!ast_moh_files_next(chan))
-			f = ast_readframe(ast_channel_stream(chan));
+	if(v){
+		if (!(ast_channel_vstream(chan) && (f = ast_readframe(ast_channel_vstream(chan))))) {
+                	if (!ast_moh_files_next(chan))
+                        	f = ast_readframe(ast_channel_vstream(chan));
+        	}
+	} else{
+		if (!(ast_channel_stream(chan) && (f = ast_readframe(ast_channel_stream(chan))))) {
+			if (!ast_moh_files_next(chan))
+				f = ast_readframe(ast_channel_stream(chan));
+		}
 	}
 
 	return f;
@@ -456,6 +473,9 @@
 
 		state->samples += f->samples;
 		state->sample_queue -= f->samples;
+		if(state->sample_queue <= 0){
+			state->sample_queue = 0;
+		}
 		if (ast_format_cmp(f->subclass.format, state->mohwfmt) == AST_FORMAT_CMP_NOT_EQUAL) {
 			ao2_replace(state->mohwfmt, f->subclass.format);
 		}
@@ -1189,6 +1209,13 @@
 		if (!S_ISREG(statbuf.st_mode))
 			continue;
 
+		//the "+1" chops off the period at the beginning	
+		ext = strrchr(filepath, '.') + 1;
+		// Set the music class as 'video' if we find a video file
+		if(ast_codec_get(ext,AST_MEDIA_TYPE_VIDEO,0)){
+			class->video = 1;
+		}
+
 		if ((ext = strrchr(filepath, '.')))
 			*ext = '\0';
 
