--- /usr/src/pjproject-2.6/pjsip/src/pjsip-ua/sip_inv.c	2016-08-30 08:40:18.000000000 +0000
+++ /usr/src/pjproject-2.6.1/pjsip/src/pjsip-ua/sip_inv.c	2018-01-23 15:41:12.425799773 +0000
@@ -35,6 +35,7 @@
 #include <pj/os.h>
 #include <pj/log.h>
 #include <pj/rand.h>
+#include <pjsip-ua/sip_config.h>
 
 /* 
  * Note on offer/answer:
@@ -1872,7 +1873,13 @@
 	    pjsip_msg_add_hdr(tdata->msg, (pjsip_hdr*)allow_hdr);
 	}
     }
-
+   
+    /* cjm-jan18 -- tryint to remove the REFER method */
+    if (PJSIP_REMOVE_REFER) {
+        const pj_str_t STR_REFER = { "REFER", 5 };
+        remove_val_from_array_hdr(allow_hdr, &STR_REFER);
+    }
+    
     /* Add Supported header */
     hdr = pjsip_endpt_get_capability(inv->dlg->endpt, PJSIP_H_SUPPORTED, NULL);
     if (hdr) {
@@ -1880,7 +1887,7 @@
 		   pjsip_hdr_shallow_clone(tdata->pool, hdr);
 	pjsip_msg_add_hdr(tdata->msg, (pjsip_hdr*)sup_hdr);
     }
-
+    
     /* Cleanup Allow & Supported headers from disabled extensions */
     cleanup_allow_sup_hdr(inv->options, NULL, allow_hdr, sup_hdr);
 
